README.md

```diff
@@ -155,6 +163,84 @@ if __name__ == '__main__':
     IOLoop.current().run_sync(main)
 ```
 
+## `notebook-http` Mode
+
+The `KernelGatewayApp.api` command line argument can be set to `notebook-http`
+to allow a properly annotated notebook to be served via HTTP endpoints. A
+notebook cell must be annotated with a single line comment at the beginning
+describing the HTTP endpoint. An example python cell is:
+
+```python
+# GET /hello/world
+print("hello world")
+```
+
+This enables an HTTP GET on `/hello/world`. The [api_intro.ipynb](https://github.com/jupyter-incubator/kernel_gateway/blob/master/etc/api_examples/api_intro.ipynb)
+provides a full example notebook for reference.
+
+### Processing requests
+
+When your cell is invoked there will be a string variable called `REQUEST`
+available in the scope of your cell's code. This string is a JSON object
+representing various properties of the request. You will need to parse this
+string to access the properties. For example, in python:
+
+```python
+# GET /hello/world
+req = json.loads(REQUEST)
+# do something with req
+```
+
+The comment syntax should be supported in every language, but it is a work in
+progress to identify all supported syntaxes.
+
+Path parameters can be specified when registering an endpoint by appending a `:`
+to a value within the path. For example, in python a path with path params
+`firstName` and `lastName` would be defined as:
+
+```python
+# GET /hello/:firstName/:lastName
+```
+
+The REQUEST object currently contains the following properties:
+
+* `body` - The JSON value of the body
+* `args` - An object with keys representing query parameter names and keys
+representing each value. The value will be an array so a query parameter can be
+specified multiple times.
+* `path` - An object of key-value pairs representing path parameters and
+their values.
+
+### Running
+The minimum number of arguments needed to run in HTTP mode are
+`--KernelGatewayApp.api=notebook-http` and
+`--KernelGatewayApp.seed_uri=some/notebook/file.ipynb`.
+
+You can run the kernel gateway in `notebook-http` mode from the Makefile:
+
+```
+make dev ARGS="--KernelGatewayApp.api='notebook-http' \
+--KernelGatewayApp.seed_uri=/srv/kernel_gateway/etc/api_examples/scotch_api.ipynb"
+```
+
+With the above Make command, all of the notebooks in `etc/api_examples` are
+mounted into `/srv/kernel_gateway/etc/api_examples/` and can be run in HTTP mode.
+
+### Setting The Response
```

Comment:
Probably should be near request too. I'm thinking *Running* either comes before all the details about how define an API in a notebook or after them all. It's still in the middle here.